tests/dataloading/test_chunked_dataset.py

```diff
@@ -0,0 +1,197 @@
+
+        for sample in x:
+            np.testing.assert_equal(sample[0], -sample[1])
+            np.testing.assert_equal(np.diff(sample[0]), 1)
+
+    def test_load_noncoincident(self, loader, sample_rate):
+        x = loader.load_noncoincident()
+        assert x.shape == (4, 2, 2 * sample_rate)
+
+        for sample in x:
+            assert (sample[0] != -sample[1]).all()
+            np.testing.assert_equal(np.diff(sample[0]), 1)
+
+    def test_iter(self, loader, sample_rate):
+        for i, x in enumerate(loader):
+            assert x.shape == (4, 2, 2 * sample_rate)
+
+            for sample in x:
+                torch.testing.assert_close(
+                    sample[0], -sample[1], rtol=0, atol=0
+                )
+                diffs = torch.diff(sample[0])
+                expected = torch.ones_like(diffs)
+                torch.testing.assert_close(diffs, expected, rtol=0, atol=0)
+        assert i == 5
+
+
+class TestChunkedDataset:
+    @pytest.fixture
+    def chunk_length(self):
+        return 2
+
+    @pytest.fixture
+    def reads_per_chunk(self):
+        return 4
+
+    @pytest.fixture
+    def chunks_per_epoch(self):
+        return 6
+
+    @pytest.fixture
+    def kernel_length(self):
+        return 1.5
+
+    @pytest.fixture
+    def batch_size(self):
+        return 8
+
+    @pytest.fixture
+    def batches_per_chunk(self):
+        return 7
+
+    @pytest.fixture(params=[0, 3])
+    def num_workers(self, request):
+        return request.param
+
+    @pytest.fixture(params=[True, False])
+    def coincident(self, request):
+        return request.param
+
+    @pytest.fixture
+    def dataset(
+        self,
+        fnames,
+        channels,
+        kernel_length,
+        sample_rate,
+        batch_size,
+        chunk_length,
+        reads_per_chunk,
+        batches_per_chunk,
+        chunks_per_epoch,
+        num_workers,
+        coincident,
+    ):
+        return ChunkedDataset(
+            sorted(fnames),
+            channels,
+            kernel_length=kernel_length,
+            sample_rate=sample_rate,
+            batch_size=batch_size,
+            reads_per_chunk=reads_per_chunk,
+            chunk_length=chunk_length,
+            batches_per_chunk=batches_per_chunk,
+            chunks_per_epoch=chunks_per_epoch,
+            num_workers=num_workers,
+            coincident=coincident,
+            pin_memory=False,
+        )
+
+    def test_init(self, dataset, num_workers):
+        assert dataset.chunk_size == 256
+        assert dataset.kernel_size == 192
+        if num_workers:
+            assert dataset.chunk_loader.dataset.reads_per_chunk == 1
+
+    def test_iter(self, dataset, coincident):
+        for i, x in enumerate(dataset):
+            assert x.shape == (8, 2, 192)
+            if coincident:
+                torch.testing.assert_close(x[:, 0], -x[:, 1], rtol=0, atol=0)
+            else:
+                assert (x[:, 0] != -x[:, 1]).any().item()
```

Comment:
I think this should be `.all()` like it is above.

Reply:
No because it's possible (though unlikely) that a segment gets paired with its coincident counterpart (though maybe it's so unlikely that it's worth failing this test one out of however many times to have a more thorough check)

Reply:
Yeah, I think better to have the more thorough check. At least, we should be consistent between this and the check on line 99.

Reply:
Yeah I think the reason I had those different was that the odds of having the chunks match up are way lower, since you're taking a smaller fraction of the full files (1/5, 1/2, and 1/3, respectively). But the odds of having two _kernels_ match up is much higher since the kernels occupy such a large fraction of the chunks. But agreed they should be consistent.

I would, however, make the argument that the `.any()` check is the correct one: once you've proved that when `coincident is True` that everything matches up, the only thing you can really require about when `coincident is False` is that at least _something_ doesn't match up. Then you know you're getting the behavior your expect. If some of them do match up, well that's still expected behavior some of the time. (So is all of them matching up I guess, but those odds are so low that I'm not gonna worry about it)

Reply:
Hmm, yeah, I'll buy that.

Reply:
Ok cool, this should be consistent now then